test/services/licences/supplementary/fetch-existing-licence-details.service.test.js

```diff
@@ -0,0 +1,160 @@
+'use strict'
+
+// Test framework dependencies
+const Lab = require('@hapi/lab')
+const Code = require('@hapi/code')
+
+const { describe, it, beforeEach } = exports.lab = Lab.script()
+const { expect } = Code
+
+// Test helpers
+const ChargeElementHelper = require('../../../support/helpers/charge-element.helper.js')
+const ChargeReferenceHelper = require('../../../support/helpers/charge-reference.helper.js')
+const ChargeVersionHelper = require('../../../support/helpers/charge-version.helper.js')
+const LicenceHelper = require('../../../support/helpers/licence.helper.js')
+
+// Thing under test
+const FetchExistingLicenceDetailsService = require('../../../../app/services/licences/supplementary/fetch-existing-licence-details.service.js')
+
+describe('Fetch Existing Licence Details Service', () => {
+  describe('when given a valid licenceId', () => {
+    let licence
+
+    describe('that has charge versions', () => {
+      beforeEach(async () => {
+        licence = await LicenceHelper.add()
+      })
+
+      describe('that are pre-sroc', () => {
```

Comment:
Could the licence setup be bumped a level? It would just read more clearly if it could.

```suggestion
describe('Fetch Existing Licence Details Service', () => {
  let licence

  describe('when given a licence ID for a licence that exists', () => {
    beforeEach(async () => {
      licence = await LicenceHelper.add()
    })
    
    describe('and that has charge versions', () => {
      describe('that are pre-sroc', () => {
```